cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -183,6 +188,53 @@ struct UTF8Transform {
   }
 };
 
+template <typename StringType, typename Derived>
+struct BinaryToBoolean {
```

Comment:
Do you think you could also make `BinaryContainsExact` use this struct?

Reply:
Good that you ask, would you say my approach is more, or less readable? I prefer the CRTP pattern. But it needs some changes to suit your needs (it needs to use non-statics, since it needs to store the pattern/table). If you prefer my approach I can refactor your code when I start on `startswith` or the regex version of your kernel.

Reply:
It's on the same level of readability. There is the complication that `BinaryContainsExact` needs options but that will be the same with `startswith` or the regex version (or the case-insensitve one).